web/pgadmin/browser/__init__.py

```diff
@@ -804,6 +818,41 @@ def set_master_password():
         if data != '':
             data = json.loads(data)
 
+    if not config.SERVER_MODE:
+        from pgadmin.model import Server
+        from pgadmin.utils.crypto import decrypt
+        desktop_user = User.query.first()
+        try:
+            if keyring.get_password(
+                    KEY_RING_SERVICE_NAME, KEY_RING_DESKTOP_USER.format(
+                        desktop_user.username)) or data['password']:
+                all_server = Server.query.all()
+                for server in all_server:
+                    if server.password and data['password']:
+                        name = KEY_RING_USERNAME_FORMAT.format(server.name,
+                                                               server.id)
+                        password = decrypt(server.password,
+                                           data['password']).decode()
+                        keyring.set_password(KEY_RING_SERVICE_NAME, name,
+                                             password)
+                        setattr(server, 'password', password)
+
+                keyring.set_password(KEY_RING_SERVICE_NAME,
+                                     KEY_RING_DESKTOP_USER.format(
+                                         desktop_user.username), 'test')
+                return form_master_password_response(
+                    existing=True,
+                    present=True,
+                    is_keyring=True
+                )
+            else:
+                return form_master_password_response(
+                    present=False,
+                    is_keyring=True
+                )
+        except Exception as e:
+            config.DISABLED_LOCAL_PASSWORD_STORAGE = True
```

Comment:
may be log the exception?

Reply:
Done

Reply:
Is the actual exception not imp?

Reply:
If the system fails to migrate the existing passwords due to keyring errors then use a master password.

Reply:
Yes, but don't you want to know the reason behind failure?